File: splearn-java/src/main/java/io/wisoft/splearn/adapter/integration/DummyEmailSender.java
package io.wisoft.splearn.adapter.integration;

import io.wisoft.splearn.application.member.required.EmailSender;
import io.wisoft.splearn.domain.shared.Email;
import org.springframework.context.annotation.Fallback;
import org.springframework.stereotype.Component;

@Component
@Fallback
public class DummyEmailSender implements EmailSender {

    @Override
    public void send(Email email, String subject, String body) {
        System.out.println("DummyEmailSender send email: " + email);
    }
}

File: splearn-typescript/src/main/adapter/integration/dummy-email-sender.ts
import { Email } from "@src/main/domain/email";

import type { EmailSender } from "@src/main/application/required/email-sender";

export class DummyEmailSender implements EmailSender {
  send(
    email: Email,
    subject: string,
    body: string,
  ): void {
    console.log(`DummyEmailSender send email: ${ email }`);
  }
}